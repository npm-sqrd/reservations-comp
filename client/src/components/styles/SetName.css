:local(.container) {
  /*border: 1px solid green;*/
  padding: 8px;
  padding-top: 15px;
}

:local(.input) {
  width: 210px;
  height: 35px;
  margin: 5px;
  font-size: 15px;
  padding-left: 10px;
}

:local(.button) {
  font-family: BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  background-color: #da3743;
  color: #FFF;
  width: 160px;
  height: 35px;
  border-radius: 2px;
  margin: 5px;
  font-size: 15px;
}

:local(.button:hover) {
  background-color: #b22c36;
  cursor: pointer;
}

:local(.showName) {
  font-family: BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  padding-left: 5px;
  color: #777;
  font-size: 20px;
  height: 35px;
  padding-top: 11px;
}
