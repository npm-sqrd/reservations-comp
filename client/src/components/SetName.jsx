import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles/SetName.css';

class SetName extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      nameVal: '',
    };
  }

  handleButtonClick() {
    if (this.state.nameVal.length > 0) {
      this.props.clickHandler(this.state.nameVal);
      this.setState({
        nameVal: '',
      });
    }
  }

  handleKeyPress(e) {
    if (e.key === 'Enter') {
      this.handleButtonClick();
    }
  }

  render() {
    const { name } = this.props;
    return (
      <div className={styles.container}>
        {name
        ? (
          <div className={styles.showName}>
            Welcome, {name}!  Let&#8217;s make a reservation!
          </div>)
        : (
          <div>
            <input
              className={styles.input}
              type="text"
              placeholder="Enter Name"
              id="nameInput"
              value={this.state.nameVal}
              onChange={(e) => { this.setState({ nameVal: e.target.value }); }}
              onKeyPress={e => this.handleKeyPress(e)}
            />
            <button
              className={styles.button}
              id="nameButton"
              onClick={() => this.handleButtonClick()}
            >
              Set Name
            </button>
          </div>)
      }
      </div>);
  }
}


SetName.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default SetName;
