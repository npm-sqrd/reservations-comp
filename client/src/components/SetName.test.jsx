import React from 'react';
import { shallow } from 'enzyme';

import SetName from './SetName';

describe('SetName Component', () => {
  test('should render correctly when no name is set', () => {
    expect(shallow(<SetName name="" clickHandler={() => {}} />)).toMatchSnapshot();
  });

  test('should render correctly when a name is set', () => {
    expect(shallow(<SetName name="Marge Simpson" clickHandler={() => {}} />)).toMatchSnapshot();
  });

  test('should set state.nameVal when input text changes', () => {
    const testName = 'Bart Simpson';
    const component = shallow(<SetName name="" clickHandler={() => {}} />);
    const input = component.find('input').first();
    input.simulate('change', { target: { value: testName } });
    expect(component.state('nameVal')).toBe(testName);
  });

  test('should send correct input to clickHandler function', () => {
    const mockClickHandler = jest.fn();
    const testName = 'Lisa Simpson';
    const component = shallow(<SetName
      name=""
      clickHandler={mockClickHandler}
    />);
    const input = component.find('input').first();
    const button = component.find('button').first();

    input.simulate('change', { target: { value: testName } });
    button.simulate('click');

    expect(mockClickHandler.mock.calls.length).toBe(1);
    expect(mockClickHandler.mock.calls[0][0]).toBe(testName);
  });

  test('should not call clickHandler function if textInput is empty', () => {
    const mockClickHandler = jest.fn();
    const testName = '';
    const component = shallow(<SetName
      name=""
      clickHandler={mockClickHandler}
    />);
    const input = component.find('input').first();
    const button = component.find('button').first();

    input.simulate('change', { target: { value: testName } });
    button.simulate('click');

    expect(mockClickHandler.mock.calls.length).toBe(0);
  });

  test('should send correct input to clickHandler function on Enter', () => {
    const mockClickHandler = jest.fn();
    const testName = 'Lisa Simpson';
    const component = shallow(<SetName
      name=""
      clickHandler={mockClickHandler}
    />);
    const input = component.find('input').first();

    input.simulate('change', { target: { value: testName } });
    input.simulate('keyPress', { key: 'Enter' });

    expect(mockClickHandler.mock.calls.length).toBe(1);
    expect(mockClickHandler.mock.calls[0][0]).toBe(testName);
  });

  test('should not call clickHandler function if textInput is empty on Enter', () => {
    const mockClickHandler = jest.fn();
    const testName = '';
    const component = shallow(<SetName
      name=""
      clickHandler={mockClickHandler}
    />);
    const input = component.find('input').first();

    input.simulate('change', { target: { value: testName } });
    input.simulate('keyPress', { key: 'Enter' });

    expect(mockClickHandler.mock.calls.length).toBe(0);
  });

  test('should not call clickHandler function if non-Enter key is pressed', () => {
    const mockClickHandler = jest.fn();
    const testName = 'Maggie Simpson';
    const component = shallow(<SetName
      name=""
      clickHandler={mockClickHandler}
    />);
    const input = component.find('input').first();

    input.simulate('change', { target: { value: testName } });
    input.simulate('keyPress', { key: 'U' });

    expect(mockClickHandler.mock.calls.length).toBe(0);
  });
});
