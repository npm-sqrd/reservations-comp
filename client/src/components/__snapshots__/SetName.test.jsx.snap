// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetName Component should render correctly when a name is set 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SetName
    clickHandler={[Function]}
    name="Marge Simpson"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
      >
        Welcome, 
        Marge Simpson
        !  Let’s make a reservation!
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Welcome, ",
          "Marge Simpson",
          "!  Let’s make a reservation!",
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        "Welcome, ",
        "Marge Simpson",
        "!  Let’s make a reservation!",
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
        >
          Welcome, 
          Marge Simpson
          !  Let’s make a reservation!
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Welcome, ",
            "Marge Simpson",
            "!  Let’s make a reservation!",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          "Welcome, ",
          "Marge Simpson",
          "!  Let’s make a reservation!",
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SetName Component should render correctly when no name is set 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SetName
    clickHandler={[Function]}
    name=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <input
          className={undefined}
          id="nameInput"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Enter Name"
          type="text"
          value=""
        />
        <button
          className={undefined}
          id="nameButton"
          onClick={[Function]}
        >
          Set Name
        </button>
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className={undefined}
            id="nameInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter Name"
            type="text"
            value=""
          />,
          <button
            className={undefined}
            id="nameButton"
            onClick={[Function]}
          >
            Set Name
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "id": "nameInput",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "Enter Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Set Name",
            "className": undefined,
            "id": "nameButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Set Name",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <input
            className={undefined}
            id="nameInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter Name"
            type="text"
            value=""
          />
          <button
            className={undefined}
            id="nameButton"
            onClick={[Function]}
          >
            Set Name
          </button>
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className={undefined}
              id="nameInput"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="Enter Name"
              type="text"
              value=""
            />,
            <button
              className={undefined}
              id="nameButton"
              onClick={[Function]}
            >
              Set Name
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "id": "nameInput",
              "onChange": [Function],
              "onKeyPress": [Function],
              "placeholder": "Enter Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Set Name",
              "className": undefined,
              "id": "nameButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Set Name",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
